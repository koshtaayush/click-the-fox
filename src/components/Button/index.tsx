//Library
import React from 'react'
import styled from 'styled-components'

interface Props {
    onClickProp: () => void
    buttonType: 'S' | 'L'
    disabled: boolean
    buttonText: string
}

const Button: React.FC<Props> = (props) => {
    const { onClickProp, buttonType, disabled, buttonText } = props
    return (
        <React.Fragment>
            <ButtonContainer
                type="button"
                name="sendMessage"
                onClick={onClickProp}
                disabled={disabled}
                buttonType={buttonType}
                test-id={'buttonTest'}>
                {buttonText}
            </ButtonContainer>
        </React.Fragment>
    )
}

export default Button

interface IButtonContainer {
    buttonType: string
    disabled: boolean
}

const ButtonContainer = styled.button<IButtonContainer>`
    background-color: ${(props) => (props.disabled ? '#fff' : '#FFC000')};
    color: ${(props) => (props.disabled ? '#000000' : '#FFFFFF')};
    height: 3.5rem;
    border: 0.125rem solid #c09003;
    border-radius: 0.25rem;
    padding: ${(props) => (props.buttonType === 'S' ? '0.5rem 1rem' : '1rem 4rem')};
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    font-weight: 500;
    font-size: 1rem;
    margin-right: 1rem;
`
