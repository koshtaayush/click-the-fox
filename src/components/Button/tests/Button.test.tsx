import React from 'react'
import { shallow } from 'enzyme'

import Button from './../index'

describe('Button', () => {
    it('Should match the snapshot', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Button
                onClickProp={mockFn}
                buttonType={'S'}
                disabled={false}
                buttonText={'Play'}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('Should render the button component correctly', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Button
                onClickProp={mockFn}
                buttonType={'S'}
                disabled={false}
                buttonText={'Play'}
            />
        )

        expect(component.find({ 'test-id': 'buttonTest' })).toHaveLength(1)
    })

    it('Should pass the same value which is passed', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Button
                onClickProp={mockFn}
                buttonType={'S'}
                disabled={false}
                buttonText={'Play'}
            />
        )

        expect(
            component.find({ 'test-id': 'buttonTest' }).props().buttonType
        ).toEqual('S')
    })

    it('Should render the proper name of button', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Button
                onClickProp={mockFn}
                buttonType={'S'}
                disabled={false}
                buttonText={'Play'}
            />
        )

        expect(component.text()).toEqual('Play')
    })
    
    it('Should call the click function on the event onClick', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Button
                onClickProp={mockFn}
                buttonType={'S'}
                disabled={false}
                buttonText={'Play'}
            />
        )

        component.find({ 'test-id': 'buttonTest'}).simulate("click")
        expect(mockFn).toHaveBeenCalled()
    })
})
