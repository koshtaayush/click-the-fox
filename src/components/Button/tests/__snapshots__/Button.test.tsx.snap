// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the snapshot 1`] = `
<Fragment>
  <styled.button
    buttonType="S"
    disabled={false}
    name="sendMessage"
    onClick={[MockFunction]}
    test-id="buttonTest"
    type="button"
  >
    Play
  </styled.button>
</Fragment>
`;
